### PR TITLE
feat: Allow ARM architecture to use SIMD trit validation

### LINKED ISSUE #39: ARM NEON SIMD instruction set support
Reference:
* [neon-guide](https://github.com/thenifty/neon-guide)

### PR DESCRIPTION
**Pull request by marktwtn**
A new project sse2neon is added as git submodule to allow dcurl running
on ARM architecture to use SIMD acceleration without writing NEON
intrinsic functions.

Close #39.

### CODE REVIEW

File: src/trinary_sse42.h [hunk 1]
```diff
@@ -9,8 +9,13 @@
 #ifndef TRINARY_SSE42_H_
 #define TRINARY_SSE42_H_
 
+#if defined(__SSE4_2__)
 #include <nmmintrin.h>
+#endif
 #include "constants.h"
+#if defined(__ARM_NEON)
+#include "sse2neon.h"
+#endif
```
Comment (jserv) [inline src/trinary_sse42.h:18]: Add `#error` when both SSE4.2 and Arm NEON are not available.

**Comment by jserv** [comment]
Can you check CI status?

**Comment by marktwtn** [comment]
> Can you check CI status?

I'll fix the **aarch64** CI testing problem later.
The **remote** testing is caused by the inactivated **RabbitMQ**, which will be recovered next week.

The pull request should not be merged until the CI testings are all passed.

File: src/trinary_sse42.h (hunk 1 above)
Comment (marktwtn) [reply src/trinary_sse42.h:18]: Done.

File: mk/submodule.mk [hunk 2]
```diff
@@ -54,11 +54,23 @@ endif
...
+
+$(SSE2NEON_PATH)/sse2neon.h:
+	git submodule update --init $(SSE2NEON_PATH)
+
 # Submodules
 SUBS := $(LIBTUV_PATH)/include
 ifeq ($(BUILD_REMOTE),1)
     SUBS += $(LIBRABBITMQ_PATH)/librabbitmq
 endif
+ifeq ($(UNAME_M),arm)
```
Comment (jserv) [inline mk/submodule.mk]: Can you merge two `ifeq` conditions? Use `-o`.

File: src/trinary_sse42.h [hunk 3]
```diff
@@ -9,8 +9,16 @@
 #ifndef TRINARY_SSE42_H_
 #define TRINARY_SSE42_H_
 
+#if defined(__SSE4_2__)
 #include <nmmintrin.h>
+#endif
```
Comment (jserv) [inline src/trinary_sse42.h:14]: Can you move `sse2neon.h` right after `nmmintrin.h`?

Comment (marktwtn) [reply src/trinary_sse42.h:14]: I would like to.
But I was thinking about the alphabetical order of the header files.

File: mk/submodule.mk [hunk 4]
```diff
@@ -69,3 +81,8 @@ SUB_INCLUDE := $(LIBTUV_INCLUDE)
 ifeq ($(BUILD_REMOTE),1)
     SUB_INCLUDE += $(LIBRABBITMQ_INCLUDE)
 endif
+ifeq ($(UNAME_M),arm)
+    SUB_INCLUDE += $(SSE2NEON_INCLUDE)
+else ifeq ($(UNAME_M),aarch64)
```
Comment (jserv) [inline mk/submodule.mk]: Ditto. Use `-o`.

File: mk/submodule.mk (hunk 2 above)
Comment (marktwtn) [reply mk/submodule.mk]: I'll merge them with `ifeq ($(UNAME_M),$(filter $(UNAME_M),arm aarch64))`.
Makefile does not support `-o` like shell script.

Comment (marktwtn) [reply mk/submodule.mk]: Done.

File: mk/submodule.mk (hunk 4 above)
Comment (marktwtn) [reply mk/submodule.mk]: Done.

File: src/trinary_sse42.h (hunk 3 above)
Comment (marktwtn) [reply src/trinary_sse42.h:14]: Since `sse2neon.h` is not a system header file, I keep it in the same place.

**Pull request merged**